internal/trace/trace.go

```diff
@@ -0,0 +1,106 @@
+package trace
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mark3labs/mcp-go/server"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// set a default tracer name
+var tracerName = "buildkite-mcp-server"
+
+func NewProvider(ctx context.Context, name, version string) (*sdktrace.TracerProvider, error) {
+	exp, err := otlptracegrpc.New(ctx)
```

Comment:
I feel like there is a way of getting it to pick the http or grpc transport from the env vars but I'm away from the computer just now.

Reply:
@catkins yeah it is complex, I have tried a few things, however I returned to similar to what is in the [buildkite/agent tracing](https://github.com/buildkite/agent/blob/fd239e03db6abffdb562ef0963459ab90203b763/internal/job/tracing.go#L128-L133) 

This creates an unconfigured GRPC otel exporter, which won't do anything without environment variables.